torch/csrc/monitor/events.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <unordered_map>
+
+#include <c10/util/variant.h>
+
+namespace torch {
+namespace monitor {
+
+using metadata_value_t = c10::variant<std::string, double, int64_t, bool>;
+
+struct Event {
```

Comment:
Per the RFC it has a type which can be used to encode both. You could make it be `torchx.RunnerEvent`

Reply:
I see, its fine if we encode it as a single variable (type) and use a well-known delimiter (".").  The issue is that if the log archiving service offers more columns (e.g type, group, name), then we are mapping type -> multiple columns. 

In this case, we'd be splitting `torch.monitor.Event.type` in the EventHandler, so there is an implicit assumption about what the type string will split into depending on the Event Handler. 

For example if type = "torch.monitor.Stat.foobar.AVG" and the event sink offers  "type", "group", and "name" columns. It could decide to log `(torch, monitor.Stat, foobar.AVG)`, or `(torch.monitor, Stat, foobar.AVG)` or (torch.monitor, stat.foobar, AVG)`, etc...

I think this is fine, but does require some "best-practice" guidelines on how one should implement an EventHandler in this case.

Reply:
added comments on how to best use these events and handler interfaces